packages/effect/src/Stream.ts

```diff
@@ -3472,6 +3472,42 @@ export const provideSomeLayer: {
   ): Stream<A, E | E2, RIn | Exclude<R, ROut>>
 } = internal.provideSomeLayer
 
+/**
+ * Returns a stream that mirrors the first upstream to emit an item.
+ * As soon as one of the upstream emits a first value, the other is interrupted.
+ * The resulting stream will forward all items from the "winning" source stream.
+ * Any upstream failures will cause the returned stream to fail.
+ *
+ * @example
+ * import { Stream, Schedule, Console, Effect } from "effect"
+ *
+ * const stream = Stream.fromSchedule(Schedule.spaced('2 millis')).pipe(
+ *   Stream.race(Stream.fromSchedule(Schedule.spaced('1 millis'))),
+ *   Stream.take(6),
+ *   Stream.tap(Console.log)
+ * )
+ *
+ * Effect.runPromise(Stream.runDrain(stream))
+ * // Output each millisecond from the first stream, the rest streams are interrupted
+ * // 0
+ * // 1
+ * // 2
+ * // 3
+ * // 4
+ * // 5
+ * @since 3.6.0
+ * @category racing
+ */
+export const race: {
+  <A2, E2, R2>(
+    stream2: Stream<A2, E2, R2>
+  ): <A1, E1, R1>(stream1: Stream<A1, E1, R1>) => Stream<A1 | A2, E1 | E2, R1 | R2>
+  <A1, E1, R1, A2, E2, R2>(
+    stream1: Stream<A1, E1, R1>,
+    stream2: Stream<A2, E2, R2>
```

Comment:
To align with the other apis:

```suggestion
    that: Stream<A2, E2, R2>
  ): <A1, E1, R1>(self: Stream<A1, E1, R1>) => Stream<A1 | A2, E1 | E2, R1 | R2>
  <A1, E1, R1, A2, E2, R2>(
    self: Stream<A1, E1, R1>,
    that: Stream<A2, E2, R2>
```